packages/office-ui-fabric-react/src/components/Button/ButtonBase.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import {
+  BaseComponent,
+  css,
+  assign,
+  getId,
+  getNativeProps,
+  buttonProperties,
+  anchorProperties
+} from '../../Utilities';
+import { IButtonProps, IButton, ButtonType } from './Button.Props';
+
+export interface IButtonState {
+  labelId?: string;
+  descriptionId?: string;
+  ariaDescriptionId?: string;
+}
+
+export abstract class ButtonBase extends BaseComponent<IButtonProps, IButtonState> implements IButton {
+  public static defaultProps: IButtonProps = {
+    buttonType: ButtonType.default
+  };
+  private _buttonElement: HTMLButtonElement;
+
+  constructor(props: IButtonProps) {
+    super(props, { 'rootProps': null });
+
+    this.state = {
+      labelId: getId('Button'),
+      descriptionId: getId('Button'),
+      ariaDescriptionId: getId('Button')
+    };
+  }
+
+  public render(): JSX.Element {
+    let { buttonType, children, icon, description, ariaLabel, ariaDescription, href, disabled, onClick } = this.props;
+    let { labelId, descriptionId, ariaDescriptionId } = this.state;
+
+    const renderAsAnchor: boolean = !!href;
+    const tag = renderAsAnchor ? 'a' : 'button';
+    const nativeProps = getNativeProps(this.props.rootProps || this.props, renderAsAnchor ? anchorProperties : buttonProperties);
+    const className = css((this.props.className));
+
+    const iconSpan = icon && (buttonType === ButtonType.command || buttonType === ButtonType.hero || buttonType === ButtonType.icon)
+      ? <span className='ms-Button-icon'><i className={ `ms-Icon ms-Icon--${icon}` }></i></span>
+      : null;
+
+    // ms-Button-description is only shown when the button type is compound.
+    // In other cases it will not be displayed.
+    const descriptionSpan: React.ReactElement<React.HTMLProps<HTMLSpanElement>> = description
+      ? <span className='ms-Button-description' id={ descriptionId }>{ description }</span>
+      : null;
+
+    // If ariaDescription is given, descriptionId will be assigned to ariaDescriptionSpan,
+    // otherwise it will be assigned to descriptionSpan.
+    const ariaDescriptionSpan: React.ReactElement<React.HTMLProps<HTMLSpanElement>> = ariaDescription
+      ? <span className='ms-u-screenReaderOnly' id={ ariaDescriptionId }>{ ariaDescription }</span>
+      : null;
+
+    // Check for ariaDescription, description or aria-describedby in the native props to determine source of aria-describedby
+    // otherwise default to null.
+    let ariaDescribedBy;
+
+    if (ariaDescription) {
+      ariaDescribedBy = ariaDescriptionId;
+    } else if (description) {
+      ariaDescribedBy = descriptionId;
+    } else if (nativeProps['aria-describedby']) {
+      ariaDescribedBy = nativeProps['aria-describedby'];
+    } else {
+      ariaDescribedBy = null;
+    }
+
+    return React.createElement(
+      tag,
+      assign(
+        {},
```

Comment:
one default I think we may want to consider is `type=button`. If the user provides it in nativeProps, cool it gets stomped but if you don't provide it it may be nice to normalize the expectation since apparently different browsers have different defaults.